src/ast/passes/semantic_analyser.cpp

```diff
@@ -554,7 +554,16 @@ void SemanticAnalyser::visit(Call &call)
 
   if (call.func == "hist") {
     check_assignment(call, true, false, false);
-    check_nargs(call, 1);
+    check_varargs(call, 1, 2);
+    if (call.vargs->size() == 1) {
+      call.vargs->push_back(new Integer(0, call.loc)); // default bits is 0
+    } else {
+      check_arg(call, Type::integer, 1);
+      const auto bits = bpftrace_.get_int_literal(call.vargs->at(1));
```

Comment:
Missing check for `std::nullopt`?

Reply:
not sure. line 559 should add the second argument in case it is not supplied, am I missing something?

Reply:
Ah, found the problem and fixed

Reply:
I think the code is still coercing the nullopts wrong. For example:

```
$ sudo ./build/src/bpftrace -e 'BEGIN { $k = 3; @ = hist(1,$k); exit(); }'
stdin:1:21-31: ERROR: hist: bits 94594394363600 must be 0..5
BEGIN { $k = 3; @ = hist(1,$k); exit(); }
```

See:
https://cpp.sh/?source=%2F%2F+Example+program%0A%23include+%3Ciostream%3E%0A%23include+%3Cstring%3E%0A%0Aint+main()%0A%7B%0A++std%3A%3Astring+name%3B%0A++std%3A%3Acout+%3C%3C+%22What+is+your+name%3F+%22%3B%0A++getline+(std%3A%3Acin%2C+name)%3B%0A++std%3A%3Acout+%3C%3C+%22Hello%2C+%22+%3C%3C+name+%3C%3C+%22!%5Cn%22%3B%0A%7D

I think you need an explicit check for nullopt

Reply:
hopefully fixed, I needed to check that the second argument to `hist()` is a compile time constant integer, for which I haven't found a helper, so I restricted to integer literal, same as it is done for `lhist()`.

I misunderstood how the `check_*()` functions work; based on their usage (typically the return value is not used) I thought they would throw an exception or set some global error variable that would further prevent other processing, but instead termination has to be explicit.